homeassistant/components/myio/__init__.py

```diff
@@ -0,0 +1,109 @@
+"""The myIO integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from myio.comms_thread import CommsThread  # pylint: disable=import-error
+
+from homeassistant.const import CONF_NAME
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import slugify
+
+from .const import CONF_REFRESH_TIME, DOMAIN, SENSOR
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [SENSOR]
+
+COMMS_THREAD = CommsThread()
+
+
+async def async_setup(hass, config):
+    """Set up the myIO-server component."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up config entry."""
+    _server_name = slugify(config_entry.data[CONF_NAME])
+
+    hass.data.setdefault(DOMAIN, {})[_server_name] = {}
+
+    # hass.data[DOMAIN][_server_name]["state"] = "Offline"
+    hass.states.async_set(f"{_server_name}.state", "Offline")
```

Comment:
Don't create an entity directly like this, use a regular sensor (or binary sensor) entity instead.